src/main/java/org/opensearch/timeseries/feature/SearchFeatureDao.java

```diff
@@ -899,15 +912,16 @@ public void getColdStartSamplesForPeriods(
             .<SearchRequest, SearchResponse>asyncRequestWithInjectedSecurity(
                 request,
                 client::search,
-                detector.getId(),
+                config.getId(),
                 client,
+                context,
                 searchResponseListener
             );
     }
 
-    private SearchRequest createColdStartFeatureSearchRequest(AnomalyDetector detector, List<Entry<Long, Long>> ranges, Entity entity) {
+    private SearchRequest createColdStartFeatureSearchRequest(Config detector, List<Entry<Long, Long>> ranges, Optional<Entity> entity) {
         try {
-            SearchSourceBuilder searchSourceBuilder = ParseUtils.generateEntityColdStartQuery(detector, ranges, entity, xContent);
+            SearchSourceBuilder searchSourceBuilder = ParseUtils.generateColdStartQuery(detector, ranges, entity, xContent);
```

Comment:
The query is still specific to entity, right?  ... but the name change is ok.

Reply:
It is not specific to entity anymore as I change entity field to Optional. People can set it to empty if it is not specific to entity.

Reply:
Did we have a separate call of this method without entity? Or just making the code more generic for future development?